File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l=0
        r=1
        largest=0
        while l<r and r<len(prices):
            temp=prices[r]-prices[l]
            if temp<0:
                l+=1
                r=l+1
                continue
            elif temp>largest:
                largest=temp
            r+=1
        return largest

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l=0, r=1;
        int maxPrice = 0;
        while(l<=r && r<prices.length){
            if(prices[l]>prices[r]){
                l=r;
            }
            else{
                maxPrice = Math.max(maxPrice,prices[r]-prices[l]);
            }
            r++;
        }
        return maxPrice;
    }
}